object-handlers.go

```diff
@@ -951,26 +951,26 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 	w.WriteHeader(http.StatusOK)
 
 	//Send whitespace character, once every 5secs, until CompleteMultipartUpload is done
-	done := make(chan bool)
-	go func() {
+	doneCh := make(chan bool)
+	go func(doneCh <-chan bool) {
 		for {
 			select {
 			case <-time.After(5 * time.Second):
 				w.Write([]byte(" "))
```

Comment:
There is a race condition where w.Write() here can happen after writeErrorResponseNoHeader()